packages/core/src/machines/actors/auth/resetPassword.ts

```diff
@@ -0,0 +1,115 @@
+import { createMachine, assign, sendUpdate } from 'xstate';
+
+import { AuthEvent, ResetPasswordContext } from '../../../types';
+import { Auth } from 'aws-amplify';
+
+export const resetPasswordActor = createMachine<
+  ResetPasswordContext,
+  AuthEvent
+>(
+  {
+    id: 'resetPasswordActor',
+    initial: 'resetPassword',
+    states: {
+      resetPassword: {
+        initial: 'edit',
+        exit: ['clearFormValues', 'clearError'],
+        states: {
+          edit: {
+            entry: sendUpdate(),
+            on: {
+              SUBMIT: 'submit',
+              CHANGE: { actions: 'handleInput' },
+            },
+          },
+          submit: {
+            entry: [sendUpdate(), 'setUsername', 'clearError'],
+            invoke: {
+              src: 'resetPassword',
+              onDone: {
+                target: '#resetPasswordActor.confirmResetPassword',
+              },
+              onError: {
+                actions: ['setRemoteError'],
+                target: 'edit',
+              },
+            },
+          },
+        },
+      },
+      confirmResetPassword: {
+        initial: 'edit',
+        exit: ['clearFormValues', 'clearError', 'clearUsername'],
+        states: {
+          edit: {
+            entry: sendUpdate(),
+            on: {
+              SUBMIT: 'submit',
+              RESEND: 'resendCode',
+              CHANGE: { actions: 'handleInput' },
+            },
+          },
+          resendCode: {
+            entry: ['clearError', sendUpdate()],
+            invoke: {
+              src: 'resetPassword',
+              onDone: { target: 'edit' },
+              onError: {
+                actions: 'setRemoteError',
+                target: 'edit',
+              },
+            },
+          },
+          submit: {
+            entry: ['clearError', sendUpdate()],
+            invoke: {
+              src: 'confirmResetPassword',
+              onDone: {
+                actions: 'clearUsername',
+                target: '#resetPasswordActor.resolved',
+              },
+              onError: {
+                actions: 'setRemoteError',
+                target: 'edit',
+              },
+            },
+          },
+        },
+      },
+      resolved: { type: 'final' },
+    },
+  },
+  {
+    actions: {
+      setRemoteError: assign({
+        remoteError: (_, event) => event.data?.message || event.data,
+      }),
+      setUsername: assign({
+        username: (context) => context.formValues.username,
+      }),
+      handleInput: assign({
+        formValues: (context, event) => {
+          const { name, value } = event.data;
+          return { ...context.formValues, [name]: value };
+        },
+      }),
+      clearFormValues: assign({ formValues: {} }),
+      clearError: assign({ remoteError: '' }),
+      clearUsername: assign({ username: undefined }),
```

Comment:
Looks like these actions are gonna be reused across all of the actors. Are we able to pull these out into their own `actions.ts` file?

Reply:
Yes! And I tried this for a bit too. Here's where xstate typing gets more finicky though because we have different context types for each actor and action types don't work so well with unions. 

We might need to use some `as any` or `as AssignAction<SignInContext, AuthEvent>`. Let's pull this into a different PR, as I want to get the structure laid down first. Trust me, I dislike dry code as much as you do 😆